core/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -328,25 +327,75 @@ private void xContentHeader(XContentBuilder builder, String key, List<String> va
     }
 
     /**
-     * Static toXContent helper method that also renders non {@link org.elasticsearch.ElasticsearchException} instances as XContent.
+     * Renders additional per exception information into the XContent
      */
-    public static void toXContent(XContentBuilder builder, Params params, Throwable ex) throws IOException {
-        ex = ExceptionsHelper.unwrapCause(ex);
-        if (ex instanceof ElasticsearchException) {
-            ((ElasticsearchException) ex).toXContent(builder, params);
+    protected void metadataToXContent(XContentBuilder builder, Params params) throws IOException {
+    }
+
+    /**
+     * Static toXContent helper method that also renders non {@link org.elasticsearch.ElasticsearchException} instances as XContent,
```

Comment:
can you clarify in the javadocs that it supports Throwable besides what you already wrote?